base/math.jl

```diff
@@ -503,7 +503,43 @@ end
 
 Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 """
-hypot(x::Number...) = sqrt(sum(abs2(y) for y in x))
+function hypot(x::Number...)
+    # compute infnorm x (modeled on generic_vecnormMinusInf(x) in LinearAlgebra/generic.gl)
+    (v, s) = iterate(x)::Tuple
+    maxabs = abs(v)
+    while true
+        y = iterate(x, s)
+        y === nothing && break
+        (v, s) = y
+        vnorm = abs(v)
+        maxabs = ifelse(isnan(maxabs) | (maxabs > vnorm), maxabs, vnorm)
+    end
+    maxabs = float(maxabs)
```

Comment:
Type instability?

Reply:
Yes. I'll see if I can fix that by determining the least common supertype of the elements of the Tuple x and promoting to that over the iterations.

Reply:
Why not using another variable?

Reply:
Using another variable for line 517 is good, but @code_warntype showed more type instability due to the variability of types within the Tuple argument `x`, which `xp = promote(x...)` fixes. I'm just about ready to commit a type-stable version.